compiler/hash-ast/src/ast.rs

```diff
@@ -730,11 +875,20 @@ pub enum Pat {
     Module(ModulePat),
     /// A tuple pattern is a collection of patterns, e.g `(1, x, 'c')`
     Tuple(TuplePat),
+    /// A list pattern, which is a collection of patterns e.g `[x, 2, y]`
     List(ListPat),
+    /// A literal pattern e.g. `c`
     Lit(LitPat),
+    /// An `or` pattern which groups multiple patterns e.g. `a | b | c`
     Or(OrPat),
+    /// A pattern that is guarded by an if statement, e.g. `x if x > 5`
     If(IfPat),
-    Ignore(IgnorePat),
+    /// Wildcard pattern, similar to a binding but it is not binded
+    /// to any member.
+    Wild(WildPat),
+    /// Similar to a [Pat::Wild], but does captures a collection of patterns,
+    /// which can be used to ignore a range of elements in a tuple or
+    /// a list.
```

Comment:
Surely this isn't similar to wild?

Reply:
It's similar in the sense that it matches all elements and doesn't bind any variable, but the only difference is that it captures multiple elements or patterns.

Reply:
yeah, and it can *only* be used in list context. I think that last one is quite important